pkgs/build-support/fetchnextcloudapp/default.nix

```diff
@@ -0,0 +1,38 @@
+{ stdenv, gnutar, findutils, fetchurl, ... }:
+{ name
+, url
+, version
+, sha256
+, patches ? [ ]
+}:
+stdenv.mkDerivation {
+  name = "nc-app-${name}";
+  inherit version patches;
+
+  src = fetchurl {
+    url = url;
+    sha256 = sha256;
+  };
+
+  nativeBuildInputs = [
+    gnutar
+    findutils
+  ];
+
+  unpackPhase = ''
+    tar -xzpf $src
+  '';
+
+  installPhase = ''
+    approot="$(dirname $(dirname $(find -path '*/appinfo/info.xml' | head -n 1)))"
```

Comment:
Is there an app which doesn't have the `appinfo` folder in the archive root?
We are [deploying several apps](https://framagit.org/markas/infrastructure/-/blob/7e65a9050750d5e46c9fc93f1ac9af65821bea49/modules/nextcloud.nix#L10) with Nix and didn't see this kind of issues.

We could then use the `fetchzip` fetcher instead.

Reply:
During testing I noticed that some apps were placed in a folder and had a bunch of non-related stuff in the zip. This removed that. (I can't remember which app is was tho)